nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestColdStorage.java

```diff
@@ -108,6 +110,60 @@ public void shouldFailWhenMovingToColdStorageDocumentWithoutContent() {
         }
     }
 
+    @Test
+    public void shouldRetrieveDocumentBlobFromColdStorage() {
+        DocumentModel documentModel = createDocument(true);
+
+        // move the blob to cold storage
+        ColdStorageHelper.moveContentToColdStorage(session, documentModel.getRef());
+
+        // retrieve, which means initiate a request to restore the blob from cold storage
+        documentModel = ColdStorageHelper.retrieveContentFromColdStorage(session, documentModel.getRef(),
+                NUMBER_OF_DAYS_OF_AVAILABILITY);
+        transactionalFeature.nextTransaction();
+        documentModel = session.getDocument(documentModel.getRef());
```

Comment:
`documentModel.refresh()` is simpler.

Reply:
You mean instead of re-fetching the document, but keeping `transactionalFeature.nextTransaction()` ?
Though it is not strictly needed since there aren't any asynchronous works behind, at some point we talked about always calling it to make sure and be consistent.

Reply:
Yes I mean `documentModel.refresh()` is a shorter way to write `documentModel = session.getDocument(documentModel.getRef())`
But if it's not needed then don't call it at all.